resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -189,6 +178,17 @@ private[spark] class KubernetesClusterSchedulerBackend(
           .withContainerPort(port._2)
           .build()
       })
+    val args = scala.collection.mutable.Buffer[String]()
+    args ++= Seq(
+      "${JAVA_HOME}/bin/java", s"-Dspark.executor.port=${executorPort.toString}",
```

Comment:
Similarly here - at least `JAVA_HOME/bin/java` but would like the Dockerfile to at least handle the specification of the main class as well. That being said it's tricky to separate out the various parts of the `java` command in this context, seeing as the VM options have to come before the main class.